docroot/modules/custom/mass_admin_pages/src/EventSubscriber/CookieSubscriber.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\mass_admin_pages\EventSubscriber;
+
+use Drupal\Core\Cache\Context\CacheContextsManager;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\PageCache\RequestPolicyInterface;
+use Drupal\Core\PageCache\ResponsePolicyInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\Core\EventSubscriber\FinishResponseSubscriber;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+
+/**
+ * Response subscriber to add cookies.
+ */
+class CookieSubscriber extends FinishResponseSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(LanguageManagerInterface $language_manager, ConfigFactoryInterface $config_factory, RequestPolicyInterface $request_policy, ResponsePolicyInterface $response_policy, CacheContextsManager $cache_contexts_manager, $http_response_debug_cacheability_headers = FALSE) {
+    parent::__construct($language_manager, $config_factory, $request_policy, $response_policy, $cache_contexts_manager, $http_response_debug_cacheability_headers);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setResponseNotCacheable(Response $response, Request $request) {
```

Comment:
Is there a benefit to overloading setResponseNotCacheable() with our cookie logic? I propose we implement onAllResponds() method. Otherwise code looks good.

Reply:
If we're just checking for admin paths, no, not really, beyond I guess just using $response directly. Updated the PR